desktop/lib/state/index.js

```diff
@@ -22,6 +22,14 @@ State.prototype.logout = function( ) {
 	this.loggedIn = false;
 };
 
+State.prototype.setLogPath = function( path ) {
+	this.logPath = path;
+};
+
+State.prototype.getLogPath = function( ) {
+	return this.logPath;
+};
```

Comment:
Would these be necessary if we moved the logging directory logic into the logging module? It doesn't look obvious that it needs to occur in `desktop/env.js` especially since this logic is specific to the logger and there's a `logger` module. Anything that relies on the logger module will pull it in and that module-init code that we write will run before anything can call it, so we still have the ability to initialize things before any logging call could be made.

Reply:
Yeah I'm not thrilled about this either, but this seemed like a reasonable choice given the alternatives. The logs path needs to be set relative to the user's `userData` directory, but the `userData` directory itself is overriden in the application's lifecyle in env.js.
 
```

/**
 * Initialize core components
 */

const state = require( './lib/state' );
const config = require( './lib/config' );
const appData = path.join( app.getPath( 'appData' ), config.appPathName );

// Initialize log directory prior to requiring any modules that log
const logPath = process.env.WP_DEBUG_LOG ? process.env.WP_DEBUG_LOG : path.join( appData, 'logs', 'wp-desktop.log' );
mkdirSync( path.dirname( logPath ), { recursive: true }, ( err ) => {
	if ( err ) {
		err.message = 'Failed to initialize log directory: ' + err.message;
		throw err;
	}
} );
state.setLogPath( logPath );

// Initialize settings
const Settings = require( './lib/settings' );

// Catch-all error handler
// We hook in very early to catch issues during the startup process
require( './app-handlers/exceptions' )();

// if app path set to asar, switch to the dir, not file
var apppath = app.getAppPath();
if ( path.extname( apppath ) === '.asar' ) {
	apppath = path.dirname( apppath );
}
process.chdir( apppath );

process.env.CALYPSO_ENV = config.calypso_config;

// Set app config path
app.setPath( 'userData', appData );

if ( Settings.isDebug() ) {
	process.env.DEBUG = config.debug.namespace;
}
```

It's definitely awkward having `lib/logger` know in advance about "where the overidden userData directory is", and it would be an easy footgun if the path to `userData` ever changes and we forget to change the path buried in `lib/logger` as well. Having this path set in env.js, next to the `userData` override, is far from perfect but makes it a little more obvious what's going on.

The prior implementation (that wasn't really implemented) seemed to have the log path set in the application's `config` ([ref](https://github.com/Automattic/wp-desktop/pull/814/files#diff-791771c30a54a61b1a179a125c3e9699L43)). This seems like a reasonable choice as well, save for the fact that the `config` values are set at buildtime, not runtime, so a path relative to `userData` in the config object would also smell a little funny.